tests/gui/base/test_sdcheckbox.py

```diff
@@ -0,0 +1,12 @@
+from PyQt5.QtWidgets import QApplication
+
+from securedrop_client.gui.base.checkbox import SDCheckBox
+
+app = QApplication([])
+
+
+def test_SDCheckBox(mocker):
+    checkbox_area = SDCheckBox()
+    checkbox_area.clicked = mocker.MagicMock()
+    checkbox_area.mousePressEvent(None)
+    checkbox_area.clicked.emit.assert_called_once_with()
```

Comment:
Although I see no immediate issue with this assertion, it is always somewhat risky to mock the object under test. When asserting that Qt signals have been emitted, mocking can be avoided by using a Qt helper called `QSignalSpy`.

When the signal is emitted, the spy will record each emission (and the arguments that were used, if any). In the test, the spy behaves _like_ a list (but is not a list! so not everything you can do with a list works).

Something along these lines would test the same as the current code, without the need to mock the object under test:

```python
from PyQt5.QtTest import QSignalSpy
from PyQt5.QtWidgets import QApplication

from securedrop_client.gui.base.checkbox import SDCheckBox

app = QApplication([])

def test_SDCheckBox_emits_clicked_signal():
  # Arrange
  checkbox_area = SDCheckBox()
  signal_emissions = QSignalSpy(checkbox_area.clicked)
  
  # Act
  checkbox_area.mousePressEvent(None)
  
  # Assert
  assert len(signal_emissions) == 1
```

There is also an example in #1388 (here is a [direct link](https://github.com/freedomofpress/securedrop-client/pull/1388/files#diff-0eb780839f71707d6ca54baf410b1087790b4656d7b0e92014db7e899af7d131R78-R95)  that may be broken when we update the PR).

Reply:
Amazing!! Thank you so much for this tip!